conpot/core/file_io.py

```diff
@@ -73,13 +40,106 @@ def _flush(self):
 
     def write_chunk(self, data):
         bytes_written = self._f.write(data)
-        logger.debug('Writing {} bytes of data to file {} - Contents {}'.format(self.chunk_size, self.name, data))
-        if not data or len(data) < self.chunk_size:
-            logger.debug('Finished writing file {}. Closing!'.format (self.name))
-            self._f.close()
+        logger.debug('Writing bytes of data to file {} - Contents {}'.format(self.name, data))
 
         return bytes_written
 
     def __del__(self):
         if self._f and not self._f.closed:
-            self._f.close()
\ No newline at end of file
+            self._f.close()
+
+
+class AbstractFS(object):
+    """
+    AbstractFS distinguishes between "real" filesystem paths and "virtual" ftp paths emulating a UNIX chroot jail
+    where the user can not escape its home directory (example: real "/home/user" path will be seen as "/" by the client)
+
+        - Use AbstractFS.home_fs object to access the chroot jail FS
+        - Use AbstractFS.root_fs object to access the global '/' FS
+
```

Comment:
Some really good stuff I thought of when I was beginning to experiment with FTP. Use `AbstractFS.root_fs` to access `'/'` of the vfs and use `AbstractFS.home_fs` to access a protocol specific chroot jail FS. `AbstractFS.home_fs` is basically `/' + 'protocol_name'` directory. But if a protocol tries to access `../` from `home_fs`, an implicit Illegal back reference exception is raised.
AbstractFS also wraps around os.* command. So you can do `AbstractFS.mkdir(path)` etc. 

Made a mistake here: Use AbstractFS.home_fs ~~object~~ attribute to access the chroot jail FS.